pkg/apis/garden/validation/validation.go

```diff
@@ -977,7 +977,7 @@ func ValidateShootUpdate(newShoot, oldShoot *garden.Shoot) field.ErrorList {
 }
 
 // ValidateShootSpec validates the specification of a Shoot object.
-func ValidateShootSpec(spec *garden.ShootSpec, fldPath *field.Path) field.ErrorList {
+func ValidateShootSpec(spec *garden.ShootSpec, mode *garden.ProxyMode, fldPath *field.Path) field.ErrorList {
```

Comment:
Why a pointer?
Also, I would rename the field (in all the various function signatures) to make it more clear which mode it is.

```suggestion
func ValidateShootSpec(spec *garden.ShootSpec, kubeProxyModeStatus garden.ProxyMode, fldPath *field.Path) field.ErrorList {
```